### ISSUE TITLE
Install on DigitalOcean fails at 'npm install' with deprecation warnings.

### ISSUE THREAD

**Issue by fabianfett** [Strider-CD/strider#913]
Hi,

I am trying to install strider on a 5buck droplet (ubuntu 14.04) at digitalocean. The prerequisites are fulfilled by:

```{r, engine='bash', count_lines}
strider@xxx:~/strider$ npm --version
2.14.12
strider@xxx:~/strider$ node --version
v4.2.6
strider@xxx:~/strider$ node-gyp --version
v3.2.1
```

I checkout strider via git and then run "npm install". The install fails with the errors:

```{r, engine='bash', count_lines}
npm WARN unmet dependency /var/strider/strider/node_modules/jade requires commander@'2.0.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/commander,
npm WARN unmet dependency which is version 2.9.0
npm WARN unmet dependency /var/strider/strider/node_modules/jade requires mkdirp@'0.3.x' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/mkdirp,
npm WARN unmet dependency which is version 0.5.1
npm WARN unmet dependency /var/strider/strider/node_modules/strider-gitlab requires async@'~0.2.9' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/async,
npm WARN unmet dependency which is version 1.5.2
npm WARN unmet dependency /var/strider/strider/node_modules/strider-gitlab requires debug@'~0.7.3' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/debug,
npm WARN unmet dependency which is version 2.2.0
npm WARN unmet dependency /var/strider/strider/node_modules/strider-gitlab requires lodash@'~2.2.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/lodash,
npm WARN unmet dependency which is version 3.10.1
npm WARN unmet dependency /var/strider/strider/node_modules/strider-gitlab requires superagent@'~1.3.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/superagent,
npm WARN unmet dependency which is version 1.7.2
npm WARN unmet dependency /var/strider/strider/node_modules/superagent requires extend@'3.0.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/extend,
npm WARN unmet dependency which is version 2.0.1
npm WARN unmet dependency /var/strider/strider/node_modules/winston requires async@'~1.0.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/async,
npm WARN unmet dependency which is version 1.5.2
npm WARN unmet dependency /var/strider/strider/node_modules/apidoc requires winston@'~2.1.1' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/winston,
npm WARN unmet dependency which is version 1.1.2
npm WARN unmet dependency /var/strider/strider/node_modules/jshint requires lodash@'3.7.x' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/lodash,
npm WARN unmet dependency which is version 3.10.1
npm WARN unmet dependency /var/strider/strider/node_modules/mocha requires commander@'2.3.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/commander,
npm WARN unmet dependency which is version 2.9.0
npm WARN unmet dependency /var/strider/strider/node_modules/mocha requires debug@'2.0.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/debug,
npm WARN unmet dependency which is version 2.2.0
npm WARN unmet dependency /var/strider/strider/node_modules/mocha requires jade@'0.26.3' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/jade,
npm WARN unmet dependency which is version 0.35.0
npm WARN unmet dependency /var/strider/strider/node_modules/mocha requires mkdirp@'0.5.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/mkdirp,
npm WARN unmet dependency which is version 0.5.1
npm WARN unmet dependency /var/strider/strider/node_modules/wd requires async@'~1.0.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/async,
npm WARN unmet dependency which is version 1.5.2
npm WARN unmet dependency /var/strider/strider/node_modules/wd requires lodash@'~3.9.3' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/lodash,
npm WARN unmet dependency which is version 3.10.1
npm WARN unmet dependency /var/strider/strider/node_modules/wd requires request@'~2.55.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/request,
npm WARN unmet dependency which is version 2.69.0
npm WARN unmet dependency /var/strider/strider/node_modules/socket.io-client/node_modules/engine.io-client requires component-emitter@'1.1.2' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/socket.io-client/node_modules/component-emitter,
npm WARN unmet dependency which is version 1.2.0
npm WARN unmet dependency /var/strider/strider/node_modules/socket.io-client/node_modules/socket.io-parser requires component-emitter@'1.1.2' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/socket.io-client/node_modules/component-emitter,
npm WARN unmet dependency which is version 1.2.0
npm WARN unmet dependency /var/strider/strider/node_modules/socket.io/node_modules/socket.io-adapter/node_modules/socket.io-parser requires debug@'0.7.4' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/debug,
npm WARN unmet dependency which is version 2.2.0
npm WARN unmet dependency /var/strider/strider/node_modules/strider-git/node_modules/fs-extra requires mkdirp@'0.3.x' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/mkdirp,
npm WARN unmet dependency which is version 0.5.1
npm WARN unmet dependency /var/strider/strider/node_modules/strider-cli/node_modules/prompt requires winston@'0.8.x' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/winston,
npm WARN unmet dependency which is version 1.1.2
npm WARN unmet dependency /var/strider/strider/node_modules/strider-cli/node_modules/rimraf requires glob@'^6.0.1' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/strider-cli/node_modules/glob,
npm WARN unmet dependency which is version 4.5.3
npm WARN unmet dependency /var/strider/strider/node_modules/strider-cli/node_modules/superagent requires debug@'~2.0.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/debug,
npm WARN unmet dependency which is version 2.2.0
npm WARN unmet dependency /var/strider/strider/node_modules/strider-cli/node_modules/superagent requires extend@'~1.2.1' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/extend,
npm WARN unmet dependency which is version 2.0.1
npm WARN unmet dependency /var/strider/strider/node_modules/strider-extension-loader/node_modules/connect requires cookie@'0.0.4' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/cookie,
npm WARN unmet dependency which is version 0.1.5
npm WARN unmet dependency /var/strider/strider/node_modules/strider-heroku/node_modules/strider-git requires mkdirp@'~0.3.5' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/mkdirp,
npm WARN unmet dependency which is version 0.5.1
npm WARN unmet dependency /var/strider/strider/node_modules/strider-heroku/node_modules/superagent requires debug@'~0.7.2' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/debug,
npm WARN unmet dependency which is version 2.2.0
npm WARN unmet dependency /var/strider/strider/node_modules/strider-python/node_modules/strider-detection-rules/node_modules/gumshoe requires async@'~0.1.22' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/async,
npm WARN unmet dependency which is version 1.5.2
npm WARN unmet dependency /var/strider/strider/node_modules/strider-ruby/node_modules/strider-detection-rules/node_modules/gumshoe requires async@'~0.1.22' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/async,
npm WARN unmet dependency which is version 1.5.2
npm WARN unmet dependency /var/strider/strider/node_modules/strider-simple-runner/node_modules/strider-runner-core requires mkdirp@'~0.3.5' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/mkdirp,
npm WARN unmet dependency which is version 0.5.1
npm WARN unmet dependency /var/strider/strider/node_modules/strider-simple-runner/node_modules/strider-runner-core requires async@'~0.2.9' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/strider-simple-runner/node_modules/async,
npm WARN unmet dependency which is version 0.1.22
npm WARN unmet dependency /var/strider/strider/node_modules/strider-simple-runner/node_modules/fs-extra requires mkdirp@'0.3.x' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/mkdirp,
npm WARN unmet dependency which is version 0.5.1
npm WARN unmet dependency /var/strider/strider/node_modules/strider-simple-runner/node_modules/dirkeeper requires async@'~0.2.9' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/strider-simple-runner/node_modules/async,
npm WARN unmet dependency which is version 0.1.22
npm WARN unmet dependency /var/strider/strider/node_modules/bower-installer/node_modules/bower requires glob@'~4.0.2' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/bower-installer/node_modules/glob,
npm WARN unmet dependency which is version 3.2.11
npm WARN unmet dependency /var/strider/strider/node_modules/bower-installer/node_modules/bower requires mkdirp@'0.5.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/bower-installer/node_modules/mkdirp,
npm WARN unmet dependency which is version 0.3.5
npm WARN unmet dependency /var/strider/strider/node_modules/bower-installer/node_modules/bower requires nopt@'~3.0.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/bower-installer/node_modules/nopt,
npm WARN unmet dependency which is version 2.1.2
npm WARN unmet dependency /var/strider/strider/node_modules/bower-installer/node_modules/bower requires request@'~2.42.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/request,
npm WARN unmet dependency which is version 2.69.0
npm WARN unmet dependency /var/strider/strider/node_modules/bower-installer/node_modules/bower requires semver@'~2.3.0' but will load
npm WARN unmet dependency /var/strider/strider/node_modules/semver,
npm WARN unmet dependency which is version 4.3.6
```

Interestingly on my local vm (ubuntu 14.04) everything works as expected: Same npm and nodejs version. Do you have any ideas what is causing this?

**Comment by knownasilya** [comment]
Maybe something to do with the network access from the droplet?

**Comment by fabianfett** [comment]
```{r, engine='bash', count_lines}
strider@xxx:~$ sudo ufw status verbose
Status: inactive
```
I also have never run into any network access problems with droplets. AFAIK they can access the internet without restrictions.

**Comment by knownasilya** [comment]
Have you tried everything here http://stackoverflow.com/a/20765400/483616

**Comment by fabianfett** [comment]
After looking more closely into the issue, I noticed that the `npm install` command got killed after some time. Digging around the web I found that this is most likely due to memory constraints. **Creating and enabling a swap file** therefore solved the issue. 

Interesting side note: Apparently 512mb RAM is not enough for an install of the git checked out version. `npm install strider` though went through smoothly though (even without added swap).
